### ISSUE TITLE
Discount on invoices

### ISSUE THREAD

**Issue by daboss84** [kevinpapst/kimai2#3349]
**Is your feature request related to a problem? Please describe.**
Sometimes, the one othe the other customer get a discount on their invoice (like 10% off the total net).

**Describe the solution you'd like**
For example in the invoice template, add a field for the discount. The discount gets calculated and added to the invoice and all other relevant parts like the invoice overview.

**Describe alternatives you've considered**
I am currently handling the discount via invoice -twig-temlates which are assigned to different invoice templates in the GUI.
Of course, this quick-and-dirty solution doesn't consider other places, where this info might be relevant like in the invoice overview, where the invoice amount without discount is displayed.

**Comment by kevinpapst** [comment]
You could use the expenses plugin and add negative amounts to the invoices.
This is not ideal, as it only allows for fixed sums and not percentages.

I assume you want a discount to be applied to the total sum of the invoice, right?

**Comment by daboss84** [comment]
Hi Kevin,

thanks for accepting the request. I want the discount to be applied to the total (net) sum (before VAT/"USt") of the invoice.

**Comment by kevinpapst** [comment]
Where would you configure the discount? On the invoice template? Customer?

**Comment by daboss84** [comment]
I can imagine both can be useful... But maybe in the invoice template, it makes little bit more sense because it's more flexible.

**Comment by daboss84** [comment]
Hm, looks more like a workaround :) And I'd prefer to configure a percentage and let the software do the calculations.... In you solution, just a fixed value is possible, if I understand correctly.

**Comment by StudioTakken** [comment]
Yes, if the discount is in a template then it will not be in the system, it only appears in the output, so it will not count in the income per year etc. In my workaround it is just a simple negative fixed rate timesheet item. 
Yes, a percentage would also be nice, but I think it should be possible per invoice, not per costumer.

**Comment by kevinpapst** [comment]
@StudioTakken the Expense plugin https://www.kimai.org/store/expenses-bundle.html supports your use-case. 
Please support further development of Kimai by buying it, instead of hacking the core and causing update problems in the long run.

**Comment by kevinpapst** [comment]
Saw it, thank you 😄 

I appreciate any dev help! But as maintainer there is at least one more side to it: most of all users are not experienced devs. They see your post, try to adjust the code (maybe it even works), a couple of months later they want to update and do a git pull, run into a merge conflict and ignore it and then post the most "weird bug reports". Which is a pain I the ass for me 😁

**Comment by StudioTakken** [comment]
Ok, I understand. Maybe this is not the place for this kind of suggestions. 
@allpeople: don't use my hack, bye the plugin (-;

**Comment by LiaraAlis** [comment]
I also have this problem and bought the expenses plugin recently.
Now, I've seen a free plugin for deducation times (https://www.kimai.org/store/keleo-deduction-time-bundle.html). What exactly is the difference between the expenses plugin and this free deducation time bundle plugin in relation to discounts for invoices?

In my case, for some customers I arrange a fixed budget for project with my customers and I guarantee to not charge more as agreed. Would it be possible to support fixed values instead of time for projects?

And is it already clear when the Expenses plugin for v2 will be released?

**Comment by kevinpapst** [comment]
@LiaraAlis All of your questions are off-topic for "invoice discounts by percentage". Open a discussion and I try to answer them.

**Comment by LiaraAlis** [comment]
@kevinpapst Yeah, you're right. Sorry. You can delete this messages if you want. :)
